### DISCUSSION TITLE
How can I detect page moved

### DISCUSSION THREAD

**Discussion by hidech** [webstudio-is/webstudio-community#56] [Q&A] [answered]
How can I detect page moved in custom code like this? But I can't import.

```
import { useEffect } from 'react';
function MyComponent() {
  useEffect(() => {
    console.log("Page moved");
  }, []);
}

```

**Comment by kof** [comment]
Can you pls elaborate first on the high level what you are trying to achieve from visitor perspective?

**Comment by hidech** [comment]
I want to initialize data from Supabase on page moved, using supabase javascript client.

**Reply by kof** [reply]
this is not a visitor perspective, this is builder perspective, I was trying to get a use case for your users, what kind of experience you are building specifically

**Comment by TrySound** [comment, marked as answer]
We landed dynamic data bindings. You can integrate supabase with server side rendering.

https://www.youtube.com/watch?v=2MrkfnjYuCo

**Reply by hidech** [reply]
Great work!
